### PR TITLE
Fix `HTTP::Cookie` parsing trailing semicolons

### PR DESCRIPTION
**Pull request by alexkutsan**
Fixes issue where HTTP::Cookie parser fails to parse HttpOnly and Secure attributes when they have trailing semicolons (e.g., 'HttpOnly;').

According to RFC 6265, semicolons are separators and empty attributes should be ignored. The original regex patterns expected exact matches without considering trailing semicolons.

Changes:
- Modified regex patterns in HTTP::Cookie::Parser to accept optional trailing semicolons and whitespace after cookie attributes
- Added comprehensive unit tests covering various edge cases:
  * HttpOnly/Secure with trailing semicolon
  * Multiple trailing semicolons
  * Whitespace with trailing semicolon
  * Complex cookies with all attributes and trailing semicolon
  * Cookies without HttpOnly but with trailing semicolon

All existing tests continue to pass, ensuring backward compatibility.

Fixes issue where real-world Set-Cookie headers like:
  'Set-Cookie: CookieName=...; Path=/; Domain=.example.com; Secure; HttpOnly;'
would incorrectly parse http_only as false instead of true.

Fixes: https://github.com/crystal-lang/crystal/issues/16327

### CODE REVIEW

File: src/http/cookie.cr [hunk 1]
```diff
@@ -280,14 +280,14 @@ module HTTP
...
-        MaxAgeAV       = /Max-Age=(?<max_age>[0-9]*)/i
-        ExpiresAV      = /Expires=(?<expires>#{SaneCookieDate})/i
+        ExtensionAV    = /(?<extension>[^\x00-\x1f\x7f]+)(?:\s*;)?/
+        HttpOnlyAV     = /(?<http_only>HttpOnly)(?:\s*;)?/i
+        SameSiteAV     = /SameSite=(?<samesite>\w+)(?:\s*;)?/i
+        SecureAV       = /(?<secure>Secure)(?:\s*;)?/i
+        PathAV         = /Path=(?<path>#{PathValue})(?:\s*;)?/i
+        DomainAV       = /Domain=\.?(?<domain>#{DomainValue})(?:\s*;)?/i
+        MaxAgeAV       = /Max-Age=(?<max_age>[0-9]*)(?:\s*;)?/i
+        ExpiresAV      = /Expires=(?<expires>#{SaneCookieDate})(?:\s*;)?/i
```
Comment (straight-shoota) [inline src/http/cookie.cr]: **suggestion:** Instead of adding this suffix match on each individual AV regex, we could push it upwards into `SetCookieString`. That keeps things all a bit simpler
This diff makes the specs pass as well:
```diff
-      SetCookieString = /^#{Regex::CookiePair}(?:;\s*#{Regex::CookieAV})*$/
+      SetCookieString = /^#{Regex::CookiePair}(?:;\s*#{Regex::CookieAV})*;?$/
```

Note that `\s*` isn't even there. I suppose that could be a missing test case for white-space before the semicolon?

Comment (alexkutsan) [reply src/http/cookie.cr]: fixed
